### PR TITLE
Workaround (trace sub-system hangs on buffer resize bug)

### PR DESCRIPTION
**Pull request by nkirsch**
After we adjust the buffer size, we explicitly call Clear() to ensure the kernel is ready.

### CODE REVIEW

File: ftrace/ftrace.go [hunk 1]
```diff
@@ -65,8 +65,8 @@ func (f *Ftrace) init() error {
 	return nil
 }
 
-func (f *Ftrace) NewEventType(name string, path string) (*EventType, error) {
-	etype, err := newEventType(f.fp, name, path)
+func (f *Ftrace) NewEventType(path string, name string) (*EventType, error) {
+	etype, err := newEventType(f.fp, path, name)
 	if err != nil {
```
Comment (nkirsch) [inline ftrace/ftrace.go]: Wait, why did these parameters get reversed?

**Review by tqorange — APPROVED** [review]
how did you figure out that:
```
f.Clear()
```
was the magic sauce?

**Comment by nkirsch** [comment]
I had some previous difficulties where unloading/reloading the module without clearing the trace buffer resulted in a panic, so I guessed that maybe earlier versions had a similar challenge when reallocating memory.

😅

File: ftrace/ftrace.go [hunk 2]
```diff
@@ -65,8 +65,8 @@ func (f *Ftrace) init() error {
 	return nil
 }
 
-func (f *Ftrace) NewEventType(name string, path string) (*EventType, error) {
-	etype, err := newEventType(f.fp, name, path)
+func (f *Ftrace) NewEventType(path string, name string) (*EventType, error) {
```
Comment (tqorange) [inline ftrace/ftrace.go]: gather is calling this as:
```
eType, err := f.NewEventType(n, "atsui/"+ev)
```
which looks like `name`, `path` -- is there a corresponding gather change too?

Comment (nkirsch) [reply ftrace/ftrace.go]: I already reverted that, not sure how it got in there.

Comment (nkirsch) [reply ftrace/ftrace.go]: If you look carefully, both the parameters and the use of them is reversed, so no net effect.

**Pull request merged**